### PR TITLE
feat(scripts/gui): allow selecting adaptive v-sync

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: Resources/Scripts/Gui/StartupScreen/ConfigViewTabs.as [hunk 1]
```diff
@@ -131,11 +131,10 @@ namespace spades {
...
                         "Uses a number representation which allows wider dynamic range during rendering process. " "Additionally, this allows color calculation whose value is in linear correspondence with actual energy, " "that is, physically accurate blending can be achieved.")));
 
-                cfg.AddRow(StartupScreenConfigCheckItemEditor(
-                    ui, StartupScreenConfig(ui, "r_vsync"), "0", "1",
-                    _Tr("StartupScreen", "V-Sync"),
+                cfg.AddRow(StartupScreenConfigSelectItemEditor(
+                    ui, StartupScreenConfig(ui, "r_vsync"), "0|1|-1",
                     _Tr("StartupScreen",
-                        "Enables frame rate synchronization.")));
+                        "V-Sync|" "Off|" "Enabled|" "Smooth")));
```
Comment (yvt) [inline Resources/Scripts/Gui/StartupScreen/ConfigViewTabs.as:137]:
```suggestion
                        "Vertical Sync:Synchronizes screen updates to a monitor's refresh rate.|" "Off|On|Adaptive")));
```

An example from another game:
Vertical Sync: Off, On, Adaptive

File: Sources/Gui/SDLRunner.cpp [hunk 2]
```diff
@@ -45,7 +45,11 @@
 SPADES_SETTING(r_videoWidth);
 SPADES_SETTING(r_videoHeight);
 DEFINE_SPADES_SETTING(r_fullscreen, "0");
+#ifdef __linux__
+DEFINE_SPADES_SETTING(r_vsync, "-1");
+#else
```
Comment (yvt) [inline Sources/Gui/SDLRunner.cpp:50]: Have you verified the impact of defaulting `r_vsync` to `-1`? The failure of `SDL_GL_SetSwapInterval` is fatal, so I would rather play safe with this.

File: Sources/Gui/SDLRunner.cpp [hunk 3]
```diff
@@ -518,7 +522,8 @@ namespace spades {
 
 #ifndef __sun
 					if (rtype == RendererType::GL) {
-						if (r_vsync != 0 && SDL_GL_SetSwapInterval(r_vsync) != 0) {
+						int vsync = r_vsync;
+						if (vsync != 0 && SDL_GL_SetSwapInterval(vsync) != 0) {
```
Comment (yvt) [inline Sources/Gui/SDLRunner.cpp:526]: Nice change but really belongs to a separate PR.

**Review by yvt — CHANGES_REQUESTED** [review]

File: Sources/Gui/SDLRunner.cpp (hunk 3 above)
Comment (devnexen) [reply Sources/Gui/SDLRunner.cpp:526]: Not a nice change but it s in fact a fix, you asked earlier if I tried out the default with -1 and because I did I found out the value were incorrect so needed the explicit cast.

File: Resources/Scripts/Gui/StartupScreen/ConfigViewTabs.as (hunk 1 above)
Comment (devnexen) [reply Resources/Scripts/Gui/StartupScreen/ConfigViewTabs.as:137]: you re right

File: Sources/Gui/SDLRunner.cpp (hunk 2 above)
Comment (devnexen) [reply Sources/Gui/SDLRunner.cpp:50]: yes see below.

Comment (yvt) [reply Sources/Gui/SDLRunner.cpp:50]: I'm not sure if you understood me. What I meant was: Are you sure the new default value `-1` is safe to use in most Linux systems? I'd hate to see new users being puzzled by an error message saying "SDL_GL_SetSwapInterval failed".

Comment (devnexen) [reply Sources/Gui/SDLRunner.cpp:50]: ah ok I only tried two distros, good point, I ll revert it then

File: Sources/Gui/SDLRunner.cpp (hunk 3 above)
Comment (yvt) [reply Sources/Gui/SDLRunner.cpp:522]: Could you revert this change as well?

**Review by yvt — APPROVED** [review]

**Review by yvt — APPROVED** [review]
